doc/CASA.rst

```diff
@@ -0,0 +1,117 @@
+algorithm becomes obsolete, the protocol can be redefined using a
+new Path Type, and another algorithm may be introduced. This
+solution assumes that all the ASes in the whole Internet can agree
+on the same cryptographic algorithms.
+
+However, ASes may want to be more flexible in the choice of the
+algorithms they support. A more sophistic approach would be that
+each AS promotes the subset of algorithms (from some globally
+defined set of algorithms) that it supports, similarly to the
+exchange of cipher suites in TLS handshakes. This means however,
+that the data plane packets need to introduce an additional field
+for each AS, which identifies the selected algorithm. For DRKey and
+EPIC, border routers would need to potentially support many
+different algorithms, which hinders scalability and performance.
+Also, border routers are likely less agile in adopting new
+cryptographic algorithms as they often contain dedicated hardware for packet
+processing.
+
+CASA can provide most of the advantages of both those two options,
+while avoiding their drawbacks: it is a tradeoff between simplicity
+and agility.
+
+Specification
+=============
+
+Description
+-----------
+In CASA, every AS chooses exactly **one algorithm** per algorithm
+category. Those categories are globally defined and can either be
+general (PRF, MAC, ...) or protocol-specific (PRF-DRKEY,
+MAC-EPIC-HVF, ...).
+
+::
+
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | MAC:          | CMAC-AES    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | PRF:          | CBC-MAC-AES |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | MAC-EPIC-HVF: | Poly1305    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | PRF-DRKEY:    | CBC-MAC-AES |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | ...           | ...         |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+The idea behind the general categories is to cover a large set of
+protocols that only need algorithm agreement between different
+entities.
+On the other hand, the protocol-specific categories serve the needs
+of protocols that have additional requirements like high
+performance.
+In case an algorithm for a protocol-specific category is announced,
+it has priority over the one specified in the general category.
+
+Those algorithms are then announced in the beacons, so that this
+information is directly available and does not need to be fetched
+(from every on-path AS) by each protocol separately.
+
+Note the asymmetry in the CASA design:
+There is in general one side with higher and one with lower
+efficiency requirements. The side that requires higher efficiency
+chooses the algorithm to be used, as it is the one that is primarily
+affected by the algorithm's performance. This mirrors the design of
+DRKey, where one side has to fetch the key (slow), and the other side
+can recompute the key on the fly (fast).
+
+Beaconing Extensions
+--------------------
+For each category, the beacon needs to be extended by a field that
```

Comment:
Explicitly mention that CASA would be carried in beacon extensions and link to a doc file